### ISSUE TITLE
Cannot draw two MGLLineStyleLayers with lineGradient.

### ISSUE THREAD

**Issue by MaximAlien** [mapbox/mapbox-gl-native-ios#355]
It's also possible to reproduce this behavior when using vanishing route line feature in Navigation iOS SDK. Ticket can be found [here](https://github.com/mapbox/mapbox-navigation-ios/issues/2488).

### Steps to reproduce

 1. Add two `MGLLineStyleLayers` on top of each other. One has to be thinner than another one, to be able to see the color difference.
 2. Add `lineGradient` property to both layers.
 3. Observe behavior.

### Expected behavior

Both `lineGradients` should be rendered.

### Actual behavior

Only lineGradient of the layer which was added first is rendered.

### Configuration

**Mapbox SDK versions:** `6.0.0`
**iOS/macOS versions:** `iOS 13.5`
**Device/simulator models:** `iPhone 11`
**Xcode version:** `Xcode 11.5`

Please see attached examples:
1. Both layers use `lineColor`.
1. First layer uses `lineGradient`, second one uses `lineColor`.
1. First layer uses `lineColor`, second one uses `lineGradient`.
1. Both layers use `lineGradient`.

As you can see in `4th` case only first gradient was rendered, it is expected that both gradients should be rendered.

<img width="395" alt="Screen Shot 2020-07-28 at 7 20 42 PM copy" src="https://user-images.githubusercontent.com/1496498/88751705-052f7e00-d10d-11ea-9fb9-559852f09f29.png">
<img width="395" alt="Screen Shot 2020-07-28 at 7 33 36 PM copy" src="https://user-images.githubusercontent.com/1496498/88751719-0d87b900-d10d-11ea-8a61-791f6485b9d7.png">
<img width="395" alt="Screen Shot 2020-07-28 at 7 37 18 PM copy" src="https://user-images.githubusercontent.com/1496498/88751808-40ca4800-d10d-11ea-9e20-4490108f5885.png">
<img width="395" alt="Screen Shot 2020-07-28 at 7 40 17 PM copy" src="https://user-images.githubusercontent.com/1496498/88751751-1d9f9880-d10d-11ea-9ecd-ff515c688fa3.png">

**Comment by zugaldia** [comment]
/cc: @cafesilencio @abhishek1508 have you observed a similar behavior for the Android implementation?

**Comment by cafesilencio** [comment]
I haven't seen this in Android. We have 3 lines representing the route line. All three are on different layers but are positioned on top of each other. All 3 have a gradient applied to them for the vanishing route line. @MaximAlien Are you sure all of your lines are on separate layers?  The gradient is applied at the layer level. It is different than the line color in that respect.

**Comment by MaximAlien** [comment]
This looks like regression which was introduced in `6.0.0`. In `5.9.0` route line is rendered correctly. Here's comparison:
<img width="395" alt="Screen Shot 2020-07-29 at 10 23 54 AM" src="https://user-images.githubusercontent.com/1496498/88833683-81ff3e00-d187-11ea-8ced-74d5f23e2aa5.png">
<img width="395" alt="Screen Shot 2020-07-29 at 10 24 47 AM" src="https://user-images.githubusercontent.com/1496498/88833695-8592c500-d187-11ea-8eab-01f0fd862cae.png">
<img width="395" alt="Screen Shot 2020-07-29 at 10 27 26 AM" src="https://user-images.githubusercontent.com/1496498/88833704-89bee280-d187-11ea-8f1d-ad2f0cd9e434.png">
<img width="395" alt="Screen Shot 2020-07-29 at 10 28 22 AM" src="https://user-images.githubusercontent.com/1496498/88833709-8c213c80-d187-11ea-98d6-8fb8628c100f.png">

**Comment by MaximAlien** [comment]
@zugaldia, that's correct. When current issue is resolved on `mapbox-gl-native-ios` side workaround on `mapbox-navigation-ios` side will be removed. This will be done in [this](https://github.com/mapbox/mapbox-navigation-ios/issues/2508) ticket.
